model/labels/regexp_test.go

```diff
@@ -147,60 +147,62 @@ func TestOptimizeConcatRegex(t *testing.T) {
 		// Simple sets alternate and concat (bar|baz is parsed as "ba[rz]").
-		{"foo|bar|baz", []string{"foo", "bar", "baz"}},
+		{"foo|bar|baz", []string{"foo", "bar", "baz"}, true},
 		// Simple sets alternate and concat and capture
-		{"foo|bar|baz|(zz)", []string{"foo", "bar", "baz", "zz"}},
+		{"foo|bar|baz|(zz)", []string{"foo", "bar", "baz", "zz"}, true},
 		// Simple sets alternate and concat and alternates with empty matches
 		// parsed as  b(ar|(?:)|uzz) where b(?:) means literal b.
-		{"bar|b|buzz", []string{"bar", "b", "buzz"}},
+		{"bar|b|buzz", []string{"bar", "b", "buzz"}, true},
 		// Skip anchors it's enforced anyway at the root.
-		{"(^bar$)|(b$)|(^buzz)", []string{"bar", "b", "buzz"}},
+		{"(^bar$)|(b$)|(^buzz)", []string{"bar", "b", "buzz"}, true},
 		// Simple sets containing escaped characters.
-		{"fo\\.o|bar\\?|\\^baz", []string{"fo.o", "bar?", "^baz"}},
+		{"fo\\.o|bar\\?|\\^baz", []string{"fo.o", "bar?", "^baz"}, true},
 		// using charclass
-		{"[abc]d", []string{"ad", "bd", "cd"}},
+		{"[abc]d", []string{"ad", "bd", "cd"}, true},
 		// high low charset different => A(B[CD]|EF)|BC[XY]
-		{"ABC|ABD|AEF|BCX|BCY", []string{"ABC", "ABD", "AEF", "BCX", "BCY"}},
+		{"ABC|ABD|AEF|BCX|BCY", []string{"ABC", "ABD", "AEF", "BCX", "BCY"}, true},
 		// triple concat
-		{"api_(v1|prom)_push", []string{"api_v1_push", "api_prom_push"}},
+		{"api_(v1|prom)_push", []string{"api_v1_push", "api_prom_push"}, true},
 		// triple concat with multiple alternates
-		{"(api|rpc)_(v1|prom)_push", []string{"api_v1_push", "api_prom_push", "rpc_v1_push", "rpc_prom_push"}},
-		{"(api|rpc)_(v1|prom)_(push|query)", []string{"api_v1_push", "api_v1_query", "api_prom_push", "api_prom_query", "rpc_v1_push", "rpc_v1_query", "rpc_prom_push", "rpc_prom_query"}},
+		{"(api|rpc)_(v1|prom)_push", []string{"api_v1_push", "api_prom_push", "rpc_v1_push", "rpc_prom_push"}, true},
+		{"(api|rpc)_(v1|prom)_(push|query)", []string{"api_v1_push", "api_v1_query", "api_prom_push", "api_prom_query", "rpc_v1_push", "rpc_v1_query", "rpc_prom_push", "rpc_prom_query"}, true},
 		// class starting with "-"
-		{"[-1-2][a-c]", []string{"-a", "-b", "-c", "1a", "1b", "1c", "2a", "2b", "2c"}},
-		{"[1^3]", []string{"1", "3", "^"}},
+		{"[-1-2][a-c]", []string{"-a", "-b", "-c", "1a", "1b", "1c", "2a", "2b", "2c"}, true},
+		{"[1^3]", []string{"1", "3", "^"}, true},
 		// OpPlus with concat
-		{"(.+)/(foo|bar)", nil},
+		{"(.+)/(foo|bar)", nil, false},
 		// Simple sets containing special characters without escaping.
-		{"fo.o|bar?|^baz", nil},
+		{"fo.o|bar?|^baz", nil, false},
 		// case sensitive wrapper.
-		{"(?i)foo", nil},
+		{"(?i)foo", []string{"FOO"}, false},
 		// case sensitive wrapper on alternate.
-		{"(?i)foo|bar|baz", nil},
-		// case sensitive wrapper on concat.
-		{"(api|rpc)_(v1|prom)_((?i)push|query)", nil},
+		{"(?i)foo|bar|baz", []string{"FOO", "BAR", "BAZ", "BAr", "BAz"}, false},
+		// mixed case sensitive.
+		{"(api|rpc)_(v1|prom)_((?i)push|query)", nil, false},
```

Comment:
I'd like to suggest some more basic tests and also tests for unsetting the case insensitivity.

```suggestion
		{"(api|rpc)_(v1|prom)_((?i)push|query)", nil, false},
		// mixed case sensitivity concatenation only without capture group.
		{"api_v1_(?i)push", nil, false},
		// mixed case sensitivity alternation only without capture group.
		{"api|(?i)rpc)", nil, false},
		// case sensitive after unsetting insensitivity.
		{"rpc|(?i)(?-i)api", []string{"rpc", "api"}, true},
		// case sensitive after unsetting insensitivity in all alternation options.
		{"(?i)((?-i)api|(?-i)rpc)", []string{"api", "rpc"}, true},
		// mixed case sensitivity after unsetting insensitivity.
		{"(?i)rpc|(?-i)api", nil, false},
```

Reply:
Good suggestions, thanks!